File: hastaneRandevu/src/main/java/com/tugra/model/Kullanici.java
package com.tugra.model;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "kullanici")
@Schema(description = "Kullanıcı Modeli")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Kullanici implements UserDetails {

    @Id
    @Column(name = "tc_no")
    private String username;

    @Column(name = "ad")
    private String ad;

    @Column(name = "soyad")
    private String soyad;

    @Column(name = "telefon_no" , unique = true)
    private String telefonNo;

    @Column(name = "email", unique = true)
    private String email;

    @Column(name = "password")
    private String password;

    @Column(name = "dogum_tarihi")
    private Date dogumTarihi;

    @Column(name = "role")
    @Enumerated(EnumType.STRING)
    private Role role;

    @OneToMany(mappedBy = "kullanici", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<RefreshToken> refreshToken;

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return List.of(new SimpleGrantedAuthority(role.name()));
    }

    @Override
    public String getPassword() {
        return password;
    }

    @Override
    public String getUsername() {
        return username;
    }

    @Override
    public boolean isAccountNonExpired() {
        return UserDetails.super.isAccountNonExpired();
    }

    @Override
    public boolean isAccountNonLocked() {
        return UserDetails.super.isAccountNonLocked();
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return UserDetails.super.isCredentialsNonExpired();
    }

    @Override
    public boolean isEnabled() {
        return UserDetails.super.isEnabled();
    }

}

File: hastaneFlutter/lib/model/Kullanici.dart
class Kullanici{

  late String _username;
  late String _ad;
  late String _soyad;
  late String _telefonNo;
  late String _email;
  late String _password;

  Kullanici(this._username, this._ad, this._soyad, this._telefonNo, this._email,
      this._password);

  String get password => _password;

  set password(String value) {
    _password = value;
  }

  String get email => _email;

  set email(String value) {
    _email = value;
  }

  String get telefonNo => _telefonNo;

  set telefonNo(String value) {
    _telefonNo = value;
  }

  String get soyad => _soyad;

  set soyad(String value) {
    _soyad = value;
  }

  String get ad => _ad;

  set ad(String value) {
    _ad = value;
  }

  String get username => _username;

  set username(String value) {
    _username = value;
  }

}